src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1594,6 +1595,18 @@ public Void caseTemplateType(TemplateType templateType) {
 
             @Override
             public Void caseNoType(NoType type) {
+              if (type.isNoResolvedType() && opts.partialInput) {
```

Comment:
Maybe add an implementation comment pointing out what's going on in here?

Reply:
done